WeatherForecast/BL/WeatherService.cs

```diff
@@ -0,0 +1,259 @@
+﻿
+using BL.Interfaces;
+using BL.Models;
+using DAL.IRepositories;
+using DAL.Models;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace BL
+{
+    public class WeatherService : IWeatherService
+    {
+        private readonly IWeatherRepository weatherRepository;
+        public static CommentContext comment;
+
+        Stopwatch st = new Stopwatch();
+
+
+        public WeatherService(IWeatherRepository weatherRepository)
+        {
+            this.weatherRepository = weatherRepository;
+        }
+
+        public void AddAsync(Weather weather)
+        {
+             weatherRepository.Add(weather);
+        }
+
+        public Task DeleteAsync(Weather weather)
+        {
+            throw new System.NotImplementedException();
+        }
+        public async Task<List<Weather>> getreport(DateTime from, DateTime to, string city)
+        {
+            return await weatherRepository.GetByDateRange(from, to, city);
+        }
+        public Task<Weather> GetByIdAsync(int id)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public async Task<double> AddWeather(string city)
+        {
+            CancellationToken cancellationToken = CancellationToken.None;
+            if (string.IsNullOrWhiteSpace(city) || city.Length == 1)
+            {
+                throw new ArgumentNullException(nameof(city));
+            }
+
+            WeatherApiModel obj = await GetCurrWeather(city, cancellationToken);
+
+            var weather = new Weather()
+            {
+                Lat = obj.coord.lat,
+                Lon = obj.coord.lon,
+                CityName = obj.name,
+                TempC = obj.main.temp
+
+            };
+            weatherRepository.Add(weather);
+            var tmpdegreesc = Math.Round(((float)obj.main.temp), 2);
+
+
+            if (tmpdegreesc > 16)
```

Comment:
it is better to use switch case for this kind of scenarios

Reply:
added switch.